test/dynamo/test_modules.py

```diff
@@ -2113,6 +2114,49 @@ def __setattr__(self, name, value):
         # causes two compilations, bc unimplemented custom setattr
         self.assertTrue(compiles_without_buffers >= 2)
 
+    def test_cache_size_limit_on_module_graph_break(self):
```

Comment:
You will need a ton more tests for various different cases, including unspecialized nn module, and weird compositions of nn modules, including tests that guard on supers and defaults before I am fully convinced this is all safe and sound.